### PR TITLE
Fix some Clippy lints

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
New Clippy version, new lints, let's fix them.

### CODE REVIEW

**Review by chansuke — APPROVED** [review]

**Comment by lshlyapnikov** [comment]
Closes: #375.

File: src/sys/unix.rs [hunk 1]
```diff
@@ -754,6 +740,7 @@ pub(crate) fn recv_from_vectored(
 
 /// Returns the (bytes received, sending address len, `RecvFlags`).
 #[cfg(not(target_os = "redox"))]
+#[allow(clippy::unnecessary_cast)] // For `IovLen::MAX as usize` (Already `usize` on Linux).
```
Comment (lshlyapnikov) [inline src/sys/unix.rs:743]: or can simply remove the cast: `IovLen::MAX as usize`
IovLen::MAX is usize, no reason to cast it.

File: src/sys/unix.rs [hunk 2]
```diff
@@ -814,6 +801,7 @@ pub(crate) fn send_to_vectored(
...
```
Comment (lshlyapnikov) [inline src/sys/unix.rs:804]: why keeping the cast? The lint is telling you that the cast is not needed.

**Review by Darksonn — APPROVED** [review]
Sorry, I missed this when you requested a review. Feel free to ping me if I miss a review.

**Pull request merged**